connexion/apis/abstract.py

```diff
@@ -174,20 +174,43 @@ def _validate_spec(self, spec):
 
     def _set_base_path(self, base_path):
         # type: (AnyStr) -> None
-        if base_path is None:
-            if self.spec_version < (3, 0, 0):
+        if self.spec_version < (3, 0, 0):
+            if base_path is None:
                 base_path = self.specification.get('basePath', '')
+                self.base_path = canonical_base_path(base_path)
             else:
-                # TODO variable subsitution in urls for oas3
+                self.base_path = canonical_base_path(base_path)
+                self.raw_spec['base_path'] = self.base_path
+                self.specification['base_path'] = self.base_path
+        else:
+            # TODO variable subsitution in urls for oas3
+            if base_path is None:
+                # read the base_path from the spec
                 servers = self.specification.get('servers', [])
                 for server in servers:
                     # TODO how to handle multiple servers in an
                     #      oas3 spec with different paths?
-                    base_path = urlparse(server['url']).path
+                    base_path = urlsplit(server['url']).path
                     break
                 else:
                     base_path = ''
-        self.base_path = canonical_base_path(base_path)
+                self.base_path = canonical_base_path(base_path)
+            else:
+                # update the raw spec with the new base_path
+                self.base_path = canonical_base_path(base_path)
+                servers = self.specification.get('servers', [])
+                for i, server in enumerate(servers):
+                    parsed = urlsplit(server['url'])
+                    self.raw_spec['servers'][i] = server = \
+                        urlunsplit(
+                            (parsed.schem, parsed.netloc, self.base_path) +
+                            ('', '')
+                        )
+                else:
+                    raise RuntimeError(
```

Comment:
Codacy seems to be complaining about this line. Maybe embed it as part of the loop, or put an earlier check to see if any server blocks where specified.